lms/static/scripts/frontend_apps/components/ContentSelector.tsx

```diff
@@ -1,45 +1,44 @@
 import { FullScreenSpinner, LabeledButton } from '@hypothesis/frontend-shared';
 import { useMemo, useState } from 'preact/hooks';
 
+import type { Book, File, Chapter } from '../api-types';
 import { useConfig } from '../config';
 import { PickerCanceledError } from '../errors';
+import type { Content } from '../utils/content-item';
 import { GooglePickerClient } from '../utils/google-picker-client';
 import { OneDrivePickerClient } from '../utils/onedrive-picker-client';
+import type { ErrorInfo } from './FilePickerApp';
 
 import BookPicker from './BookPicker';
 import JSTORPicker from './JSTORPicker';
 import LMSFilePicker from './LMSFilePicker';
 import URLPicker from './URLPicker';
 
-/**
- * @typedef {import('../api-types').Book} Book
- * @typedef {import('../api-types').File} File
- * @typedef {import('../api-types').Chapter} Chapter
- *
- * @typedef {import('./FilePickerApp').ErrorInfo} ErrorInfo
- *
- * @typedef {'blackboardFile'|'canvasFile'|'d2lFile'|'jstor'|'url'|'vitalSourceBook'|null} DialogType
- *
- * @typedef {import('../utils/content-item').Content} Content
- */
+type DialogType =
+  | 'blackboardFile'
+  | 'canvasFile'
+  | 'd2lFile'
+  | 'jstor'
+  | 'url'
+  | 'vitalSourceBook'
+  | null;
 
-/**
- * @typedef ContentSelectorProps
- * @prop {DialogType} [defaultActiveDialog] - Used for testing
- * @prop {(ei: ErrorInfo) => void} onError
- * @prop {(c: Content) => void} onSelectContent
- */
+type ContentSelectorProps = {
```

Comment:
This type should be exported.

Reply:
Why is it? I don't see it being used outside of this component, but maybe I'm missing something 🤔

Reply:
It's our convention to export the prop typing for components as they are part of the API of the module, if that makes sense. See other component conversions for examples.

Reply:
Got it.